lib/archethic/mining/pending_transaction_validation.ex

```diff
@@ -723,8 +669,126 @@ defmodule Archethic.Mining.PendingTransactionValidation do
     end
   end
 
+  defp do_accept_transaction(%Transaction{type: :contract, data: %TransactionData{code: code}}, _) do
```

Comment:
You could just use pattern matching to test if code is empty `code: ""`. It will be more efficient than calculating the byte_size of the string. We need to know if the string is empty or not, don't need to know the size here